test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Utility/NuGetSearchServiceReconnectorTests.cs

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.ServiceHub.Framework;
+using Microsoft.VisualStudio.Threading;
+using Moq;
+using NuGet.PackageManagement.UI.Utility;
+using NuGet.VisualStudio.Internal.Contracts;
+using Xunit;
+
+namespace NuGet.PackageManagement.UI.Test.Utility
+{
+    public class NuGetSearchServiceReconnectorTests
+    {
+        private readonly JoinableTaskFactory _jtf;
+
+        public NuGetSearchServiceReconnectorTests()
+        {
+#pragma warning disable VSSDK005 // Avoid instantiating JoinableTaskContext
+            var jtc = new JoinableTaskContext();
+#pragma warning restore VSSDK005 // Avoid instantiating JoinableTaskContext
+            _jtf = new JoinableTaskFactory(jtc);
+        }
+
+        [Fact]
+        public async Task CreateAsync_NullArguments_ThrowsArgumentNullException()
+        {
+            // Arrange
+            var serviceBroker = new Mock<IServiceBroker>();
+
+            // Act & Assert
+            ArgumentNullException exception = await Assert.ThrowsAsync<ArgumentNullException>(() => NuGetSearchServiceReconnector.CreateAsync(serviceBroker: null, _jtf, CancellationToken.None));
+            Assert.Equal("serviceBroker", exception.ParamName);
+
+            exception = await Assert.ThrowsAsync<ArgumentNullException>(() => NuGetSearchServiceReconnector.CreateAsync(serviceBroker.Object, jtf: null, CancellationToken.None));
+            Assert.Equal("jtf", exception.ParamName);
+        }
+
+        [Fact]
+        public async Task Object_NormalCreation_GetInstanceFromServiceBroker()
```

Comment:
Consider a more specific name. This test is really about 1 call to `ContinueSearchAsync`, right?

Reply:
No, it's testing the instance that `ManagedNuGetSearchService` is using, and therefore `NuGetSearchService.Reconnector._service`, is the correct instance. However, since those all all private, the test validates by mocking `INuGetSearchService`, passing that instance through `IServiceBroker`, and validating that calling a method calls the correct instance. 

In other words, it's testing that the wrapping works correctly. The method being called is just coincidental because there isn't a better way to validate other than calling something.